Automated Smoketest: :x:
Sha: 052164b993065f479591797296fc835d20b10a7b
Build: failed
Number of Error(s): 9
[  0%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/CoreErrors.cpp.o
[100%] Building CXX object aws-cpp-sdk-sqs/CMakeFiles/aws-cpp-sdk-sqs.dir/source/SQSErrors.cpp.o
/mnt/disk1/home/vamosax/smoketest/PR-10318/HPCC-Platform/esp/platform/espcontext.cpp:173:34: error: conflicting return type specified for \xe2\x80\x98virtual const MemoryBuffer* CEspContext::querySessionToken()\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-10318/build/generated/esp.hpp:84:34: error:   overriding \xe2\x80\x98virtual const MemoryBuffer& IEspContext::querySessionToken()\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-10318/HPCC-Platform/esp/platform/espcontext.cpp:182:34: error: conflicting return type specified for \xe2\x80\x98virtual const MemoryBuffer* CEspContext::querySignature()\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-10318/build/generated/esp.hpp:86:34: error:   overriding \xe2\x80\x98virtual const MemoryBuffer& IEspContext::querySignature()\xe2\x80\x99
make[2]: *** [esp/bindings/SOAP/soaplib/CMakeFiles/soaplib.dir/__/__/__/platform/espcontext.cpp.o] Error 1
make[1]: *** [esp/bindings/SOAP/soaplib/CMakeFiles/soaplib.dir/all] Error 2
make: *** [all] Error 2
Get ECLWatch build result

--- a/dali/base/dasess.cpp
+++ b/dali/base/dasess.cpp
@@ -2021,13 +2021,13 @@ public:
         decrypt(buf,passwordenc);
         return buf;
     }
-    const MemoryBuffer &querySessionToken()
+    const MemoryBuffer *querySessionToken()
     {
-        return sessionToken;
+        return &sessionToken;
     }
-    const MemoryBuffer &querySignature()
+    const MemoryBuffer *querySignature()
     {
-        return signature;
+        return &signature;
     }
     virtual void set(const char *name,const char *password)
     {
@@ -2036,7 +2036,7 @@ public:
         encrypt(buf,password);
         passwordenc.set(buf.str());
     }
-    void set(const char *_name, const char *_password, const MemoryBuffer &_sessionToken, const MemoryBuffer &_signature)
+    void set(const char *_name, const char *_password, const MemoryBuffer *_sessionToken, const MemoryBuffer *_signature)
     {
         set(_name, _password);
         sessionToken.clear().append(_sessionToken);

--- a/dali/base/dasess.hpp
+++ b/dali/base/dasess.hpp
@@ -75,10 +75,10 @@ interface IUserDescriptor: extends serializable
 {
     virtual StringBuffer &getUserName(StringBuffer &buf)=0;
     virtual StringBuffer &getPassword(StringBuffer &buf)=0;
-    virtual const MemoryBuffer &querySignature()=0;//user's digital signature
-    virtual const MemoryBuffer &querySessionToken()=0;//ESP session token
+    virtual const MemoryBuffer *querySignature()=0;//user's digital signature
+    virtual const MemoryBuffer *querySessionToken()=0;//ESP session token
     virtual void set(const char *name,const char *password)=0;
-    virtual void set(const char *name,const char *password, const MemoryBuffer &_sessionToken, const MemoryBuffer &_signature)=0;
+    virtual void set(const char *name,const char *password, const MemoryBuffer *_sessionToken, const MemoryBuffer *_signature)=0;
     virtual void clear()=0;
     virtual void serializeExtra(MemoryBuffer &tgt)=0;
     virtual void deserializeExtra(MemoryBuffer &src)=0;

--- a/esp/platform/espcontext.cpp
+++ b/esp/platform/espcontext.cpp
@@ -170,18 +170,18 @@ public:
         if (m_user)
             m_user->credentials().setSessionToken(&token);
     }
-    virtual const MemoryBuffer & querySessionToken()
+    virtual const MemoryBuffer * querySessionToken()
     {
-        return m_user->credentials().getSessionToken();
+        return m_user ? &m_user->credentials().getSessionToken() : nullptr;
     }
     virtual void setSignature(const MemoryBuffer & signature)
     {
         if (m_user)
             m_user->credentials().setSignature(&signature);
     }
-    virtual const MemoryBuffer & querySignature()
+    virtual const MemoryBuffer * querySignature()
     {
-        return m_user->credentials().getSignature();
+        return m_user ? &m_user->credentials().getSignature() : nullptr;
     }
     virtual void setRealm(const char* realm)
     {

--- a/esp/scm/esp.ecm
+++ b/esp/scm/esp.ecm
@@ -86,9 +86,9 @@ interface IEspContext : extends IInterface
     virtual void setUser(ISecUser * user) = 0;
     virtual ISecUser * queryUser() = 0;
     virtual void setSessionToken(const MemoryBuffer & token) = 0;
-    virtual const MemoryBuffer & querySessionToken() = 0;
+    virtual const MemoryBuffer * querySessionToken() = 0;
     virtual void setSignature(const MemoryBuffer & signature) = 0;
-    virtual const MemoryBuffer & querySignature() = 0;
+    virtual const MemoryBuffer * querySignature() = 0;
     virtual void setResources(ISecResourceList * rlist) = 0;
     virtual ISecResourceList * queryResources() = 0;
     virtual void setSecManger(ISecManager * mgr) = 0;

